fecgov/fec-cms
# 5681-remove unused deps and pins

## Summary (required)

- Resolves #5681 

Removes unused dependencies and removes pins for dependencies that were added due to vulnerabilities that are no longer needed. Does not change current vulnerabilities very much, follow-up work needs to be done on current NPM vulnerabilities 


**Flagged as not used by npm-check**

1. acorn -is brought in by several packages, doesn’t need to be [pinned](https://github.com/fecgov/fec-cms/pull/2733)
2. ajv –is brought in by webpack, was [pinned for vuln ](https://www.google.com/url?q=https://github.com/fecgov/fec-cms/pull/3922&sa=D&source=docs&ust=1684348929726773&usg=AOvVaw0zGKX7tV34-D2-yemj9f22)
3. concat-stream -is brought in by several packages, doesn’t need to be [pinned](https://github.com/fecgov/fec-cms/pull/2733)
4. debug -is brought in by several packages, doesn’t need to be [pinned](https://github.com/fecgov/fec-cms/pull/2733)
5. lodash -is brought in by several packages, doesn’t need to be [pinned](https://github.com/fecgov/fec-cms/pull/2733)
6. ua-parser -js is brought in by draftail doesn’t need to be [[pinned](https://github.com/fecgov/fec-cms/pull/2733)](https://www.google.com/url?q=https://github.com/fecgov/fec-cms/pull/4508&sa=D&source=docs&ust=1684355142253468&usg=AOvVaw3x2nYPEv1emi3iMFTypCVQ)
7. braces -brought in by several packages, currently brought in in two versions, can be removed
8. watchify -brought in by karma-browserify doesn’t need to be [pinned](https://github.com/fecgov/fec-cms/pull/2733)
9. fbjs -brought in by draftail doesn't need to be pinned
10. Minimatch -is brought in by several packages, doesn’t need to be [pinned](https://github.com/fecgov/fec-cms/pull/2733)
11. Tough-cookie -unused 
12. Tunnel-agent -unused
13. babel/core -brought in by babel-jest
14. eslint-loader -unused
15. vinyl-buffer -unused


### Required reviewers
1-2 devs

## Impacted areas of the application

General components of the application that this PR will affect:

-  package.json

## How to test

Run npm i && npm run build on develop (you should see 77 vulnerabilities)

- pyenv virtualenv 3.9.16 pyenv5681
- pyenv activate pyenv5681
- pip install -r requirements.txt
- npm i && npm run build (you should see 76 vulnerabilities)
- pytest
- npm run test-single
Before and After: ![Screen Shot 2023-05-17 at 4 02 30 PM](https://github.com/fecgov/fec-cms/assets/66386084/9998cf6a-bf3e-4b08-8dd8-fabe54d0cc69)
- npm run lint
- npm run build-production
- npm run format
- cd fec
- python manage.py runserver
deploy to feature
- [ ] No missing features or functionality
   - [ ] Wagtail admin
   - [ ] /home/ pages (/about/ etc)
   - [ ] /data pages
   - [ ] /legal pages
   - [ ] Interactive elements (text prediction search, filters, maps, widgets, datatables, etc)
